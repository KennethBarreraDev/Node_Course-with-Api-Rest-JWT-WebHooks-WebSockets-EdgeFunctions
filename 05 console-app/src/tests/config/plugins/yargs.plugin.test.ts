//import {yarg} from '../../../config/plugins/yargs.plugin'
const runCommand = async (args: string[]) => {
    process.argv = [...process.argv, ...args]
    const { yarg } = await import('../../../config/plugins/yargs.plugin')
    return yarg
}

describe('test yargs.plugin', () => {
    const originalArgv = process.argv
    beforeEach(() => {
        process.argv = originalArgv
        jest.resetModules()
    })

    test('Should return default', async () => {
        const argv = await runCommand(['-b', '5'])

        expect(argv).toEqual(
            expect.objectContaining(
                {
                    b: 5,
                    l: 10,
                    s: false,
                    n: 'table',
                    d: 'files'
                }
            )
        )
    })

    test('Should return configuration with custom values', async () => {
        const argv = await runCommand(['-b', '10', '-l', '100', '-s', 'true', '-n', 'file', '-d', 'logs'])
        expect(argv).toEqual(
            expect.objectContaining(
                {
                    b: 10,
                    l: 100,
                    s: true,
                    n: 'file',
                    d: 'logs'
                }
            )
        )
    })
})