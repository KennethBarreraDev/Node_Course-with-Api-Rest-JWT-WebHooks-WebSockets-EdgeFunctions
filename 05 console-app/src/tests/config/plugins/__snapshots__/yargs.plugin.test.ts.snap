// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test yargs.plugin Should return default 1`] = `
{
  "$0": "node_modules\\jest-worker\\build\\workers\\processChild.js",
  "_": [],
  "b": 5,
  "base": 5,
  "d": "files",
  "destination": "files",
  "l": 10,
  "limit": 10,
  "n": "table",
  "name": "table",
  "s": false,
  "show": false,
}
`;
